src/plugins/discover/public/application/main/components/top_nav/discover_topnav.tsx

```diff
@@ -174,7 +174,7 @@ export const DiscoverTopNav = ({
 
   const dataViewPickerProps = {
     trigger: {
-      label: indexPattern?.title || '',
+      label: indexPattern?.getName() || '',
```

Comment:
I'm testing on the cloud deployment and I wonder about the title of the dataViewPicker, should be **test** in this case, right? 
![Discover_-_Elastic](https://user-images.githubusercontent.com/463851/172559470-d9cdfe46-c9b4-4202-80a0-b456be5d3e60.png)

Reply:
What's more, if I understand correctly `getName` should return **name** or **title**, so `|| ''` should be redundant, right?

Reply:
title is the name of the index pattern, in this case `elastic-*`, but the name is a friendly human readable identifier (`test`).

I have fixed the redundancy.

Reply:
ok, but why in this case `elastic-*`used instead of `test`, added a data view named `test` with title `elastic-*` 
... not a blocker of course ... BTW we should migrated `title` to a property name that is self explaining one day :)